src/sgame/sg_bot_util.cpp

```diff
@@ -317,35 +317,36 @@ void BotGetDesiredBuy( gentity_t *self, weapon_t *weapon, upgrade_t *upgrades, i
 	int credits = self->client->ps.persistant[PERS_CREDIT];
 	int usableCapital = credits + equipmentPrice;
 
+	//TODO: ensure that adding usableCapital + 1 several times can never overflow
+	//  (overflow an int by summing less than 10 values below than 2000
+	//  is more than unlikely though, but that code is not future-proof)
+	auto CanBuyWeapon = [usableCapital](weapon_t  item) { return BG_WeaponUnlocked(  item ) ? BG_Weapon(  item )->price : usableCapital + 1; };
```

Comment:
Sorry I wasn't paying attention the first time, and assumed these were returning booleans rather than an amount of money... As it is, the functions are misleadingly named, and the logic seems more contorted compared to the original. What's wrong with the original code? The conditions in the original code seem most straightforward.

Reply:
I can change the name. WeaponCost or UpgradeCost would also lie, somehow, since it would return a value which is not real when the item is not unlocked (yet).

What I am trying to improve here is the fact that, to check for any combination, you must check 2 things per equipment, and I suspect that there is even a bug here when one of those equipment is forbidden by configuration, because otherwise you probably need to add a 3rd check for stuff which can be disabled (actually only weapons).
So, to check for a combination like: pulse rifle + medium armor + radar, current code checks for 6 conditions (paired), should probably check for 7 (since it should check if weapon is authorized I think) and if someday radar, jetpack, grenades, armors are added to the list of equipment which can be forbidden (why not?) it would require 3 checks per item. In previous case, that's a total of 9 checks.
Checks which are multiplied by the number of combinations.

I don't think the changes I propose are ideal, far from it (would be better to have some sort of encapsulation somewhere else), but I believe they improve the situation and makes further changes easier to do, more obvious.

Reply:
The concept of the code is to buy a given level of armor/upgrades only if there will be enough money left over to buy a reasonably good gun. The bsuit+painsaw is performed to ensure that if the bot buys a bsuit, it will be able to buy something *at least* as expensive as a painsaw. So I think the code is working as intended. 

There is one possible scenario that could be considered sort of a bug - let's say, all guns cheaper than pulse rifle are disabled, and the bot has more money than cost(bsuit) + cost(painsaw) and less money than cost(bsuit) + cost(pulse rifle). Then the bot will end up with bsuit+rifle. In that case it might be better to not buy the bsuit unless it can also afford the pulse rifle.

I don't think this change improves the situation as it just obfuscates the code without changing functionality.

Reply:
Ok, I'll remove those changes from the patch, then.